apps/omg_watcher_info/lib/omg_watcher_info/api/transaction.ex

```diff
@@ -77,19 +107,53 @@ defmodule OMG.WatcherInfo.API.Transaction do
   Given order finds spender's inputs sufficient to perform a payment.
   If also provided with receiver's address, creates and encodes a transaction.
   """
-  @spec create(UtxoSelection.order_t()) :: UtxoSelection.advice_t()
+  @spec create(order_t()) :: create_t()
   def create(order) do
-    utxos = DB.TxOutput.get_sorted_grouped_utxos(order.owner)
-    UtxoSelection.create_advice(utxos, order)
+    with {:ok, inputs} <-
+           order.owner
+           |> DB.TxOutput.get_sorted_grouped_utxos()
+           |> select_inputs(order) do
```

Comment:
this is a single with statement which means a `case` is prefered. `with` - `Used to combine matching clauses.`